Size comparison

Cawllec opened the pull request:
## Goal

Add an extra testing stage that outputs an Android sizing table to a pull request using the Danger tool.

fractalwrench commented on tests/features/fixtures/minimalapp/.gitignore:
```diff
@@ -0,0 +1,13 @@
+*.iml
+.gradle
+/local.properties
+/.idea/caches
```
`.idea` should be sufficient to ignore any additional files that intellij might generate in future

fractalwrench commented on tests/features/fixtures/minimalapp/app/bugsnag.template:
```diff
@@ -0,0 +1,47 @@
+apply plugin: 'com.android.application'
+apply plugin: 'com.bugsnag.android.gradle'
```
It might be possible to depend on bugsnag conditionally by passing a command line argument when making the build. this would avoid the scenario where the dependency versions of something like `com.android.support:appcompat` gets out of sync between the two versions.

  https://docs.gradle.org/current/userguide/build_environment.html#sec:configuring_task_using_project_properties

fractalwrench commented on tests/features/fixtures/minimalapp/app/base.template:
```diff
@@ -0,0 +1,33 @@
+    }
+    buildTypes {
+        release {
+            minifyEnabled false
```
If it's not too much extra effort, it would also be quite useful information to know what the size difference is when an APK is minified.

fractalwrench commented on tests/features/fixtures/minimalapp/app/build.gradle:
```diff
@@ -0,0 +1,55 @@
+}
+
+if (project.hasProperty("withBugsnag")) {
+    println("Building with Bugsnag")
```
`project.logger.info` might be more appropriate if a message needs to be logged here as it will show up in any build logs on CI

fractalwrench commented on tests/features/fixtures/minimalapp/Dangerfile:
```diff
@@ -0,0 +1,95 @@
+markdown(%Q{
+  ### Android notifier sizes
+
+  | Format      | Size (kB)                                     | Minified Size (kB)                                |
```
Could we rename the Size/Minified Size titles? This made me think that it was the size of the total APK, rather than the impact of bugsnag-android. Maybe "size impact of bugsnag-android"?

Cawllec commented:
Addressed the above comments, unfortunately the `You used 'puts'` notice is due to the use of `outputs` in the directory strings, not because anything is being logged.
